### PR TITLE
fix: devtools: main.go

### PR DESCRIPTION
**Pull request by bhaskarvilles**
This version of the code separates the different responsibilities into separate functions, and uses the log package instead of `os.Exit()` and `panic()` to handle errors. Additionally, it uses the ioutil package instead of os package to read the file, which simplify the code and it also returns errors instead of panicking on them, making it easier to handle them at a higher level in the application.

### CODE REVIEW

File: venus-devtool/inline-gen/main.go [hunk 1]
```diff
@@ -20,109 +20,108 @@ const (
...
-				}).Parse(strings.Join(templateLines, "\n"))
-				if err != nil {
-					fmt.Printf("%s:%d: parsing template: %s\n", path, ln, err)
-					os.Exit(1)
-				}
+    db, err := ioutil.ReadFile(os.Args[2])
+    if err != nil {
+        log.Fatalf("Error reading file: %v", err)
+    }
+    var data map[string]interface{}
```
Comment (simlecode) [inline venus-devtool/inline-gen/main.go:27]: Need to pass `data` to `processFile`

File: venus-devtool/inline-gen/main.go [hunk 2]
```diff
@@ -20,109 +20,108 @@ const (
...
+            "add": func(a, b float64) float64 {
+                return a + b
+            },
+        }).Parse(strings.Join(templateLines, "\n"))
+        if err != nil {
+            return fmt.Errorf("parsing template: %v", err)
+        }
+
+        var b bytes.Buffer
+        err = tpl.Execute(&b, data)
```
Comment (simlecode) [inline venus-devtool/inline-gen/main.go:79]: `data` not defined

File: venus-devtool/inline-gen/main.go [hunk 3]
```diff
@@ -20,109 +20,108 @@ const (
...
+            }
+		case stateTemplate:
+			outLines = append(outLines, line)
+			if strings.TrimSpace(line) == /* inline-gen start */ {
+				state = stateGen
+continue
+}
+templateLines = append(templateLines, line)
+		case stateGen:
+			if strings.TrimSpace(line) != /* inline-gen end */ {
```
Comment (simlecode) [inline venus-devtool/inline-gen/main.go:116]: use `/* inline-gen start */`

File: venus-devtool/inline-gen/main.go [hunk 4]
```diff
@@ -20,109 +20,108 @@ const (
...
+    for i, line := range lines {
+        switch state {
+        case stateGlobal:
+            outLines = append(outLines, line)
+            if strings.TrimSpace(line) == `/* inline-gen template` {
+                state = stateTemplate
+            }
+		case stateTemplate:
+			outLines = append(outLines, line)
+			if strings.TrimSpace(line) == /* inline-gen start */ {
```
Comment (simlecode) [inline venus-devtool/inline-gen/main.go:110]: use `/* inline-gen start */`

**Comment by simlecode** [comment]
Thank you for your PR!

**Comment by bhaskarvilles** [comment]
> Thank you for your PR!

Can you fix this for me, I'm in a medical emergency?? I'll make you as collaborator.

**Comment by simlecode** [comment]
> > Thank you for your PR!
> 
> Can you fix this for me, I'm in a medical emergency?? I'll make you as collaborator.

yea, I can, please make me as collaborator.

**Review by simlecode — APPROVED** [review]
LGTM

**Comment by bhaskarvilles** [comment]
Hey @simlecode This contribution eligible for Filecoin Community PR Rewards program ?

**Comment by Joss-Hua** [comment]
hi, @bhaskarvilles, sorry there is no reward for single behavior, but do you want to become a long-term contributor to the Venus community and get some rewards?

**Comment by bhaskarvilles** [comment]
> hi, @bhaskarvilles, sorry there is no reward for single behavior, but do you want to become a long-term contributor to the Venus community and get some rewards?

Yes, can you give me more details please?

**Comment by Joss-Hua** [comment]
> > hi, @bhaskarvilles, sorry there is no reward for single behavior, but do you want to become a long-term contributor to the Venus community and get some rewards?
> 
> Hey @Joss-Hua , can you give me more details please?

What is your filecoin slack handle? I'll contact you.

**Comment by bhaskarvilles** [comment]
> > > hi, @bhaskarvilles, sorry there is no reward for single behavior, but do you want to become a long-term contributor to the Venus community and get some rewards?
> > 
> > Hey @Joss-Hua , can you give me more details please?
> 
> What is your filecoin slack handle? I'll contact you.

My handle is bhaskarvilles, I'm also in the venus fil channel also

**Pull request merged**